coding-guides/starboard.md

```diff
@@ -0,0 +1,154 @@
+
+Then, we preform a couple of checks on the reaction and the message. First, we check if the reaction is **NOT** the unicode star emote. Next, we preform two checks on the message, checking to see if the user who added the reaction authored the message, and if the message author is a bot. If neither of these checks return true, we're good to move on.
+
+```js
+module.exports = class {
+  constructor(client) {
+    this.client = client;
+  }
+
+async run(reaction, user) {
+  const message = reaction.message;
+  if (reaction.emoji.name !== '⭐') return;
+  if (message.author.id === user.id) return message.channel.send(`${user}, you cannot star your own messages.`);
+  if (message.author.bot) return message.channel.send(`${user}, you cannot star bot messages.`);
+  const { starboardChannel } = this.client.settings.get(message.guild.id);
+  }
+}
+```
+
+Now, this next block may look real complex, but it really isn't. Let's break it down.
+
+First, we declare 6 variables here. `fetch` will fetch the last 100 messages in the starboard channel, `stars` attempts to find an embed who's footer ends with the message's id.
+
+Next, we'll start an if statement for if there is already an embed in the starboard channel for this message, and we also define our final 4 variables. `star` is a simple regex to check how many stars the embed already has, `foundStar` allows us to use the color of the pre-existing embed, `image` checks if there is anything attached to the message, and embed is our new embed!
+
+I told you it wasn't that complicated. Let's keep going.
+
+```js
+const fetch = await message.guild.channels.find('name', starboardChannel).fetchMessages({ limit: 100 });
+const stars = fetch.find(m => m.embeds[0].footer.text.startsWith('⭐') && m.embeds[0].footer.text.endsWith(message.id));
+if (stars) {
+  const star = /^\⭐\s([0-9]{1,3})\s\|\s([0-9]{17,20})/.exec(stars.embeds[0].footer.text);
+  const foundStar = stars.embeds[0];
+  const image = message.attachments.size > 0 ? await this.extension(reaction, message.attachments.array()[0].url) : '';
+  const embed = new RichEmbed()
+    .setColor(foundStar.color)
+    .setDescription(foundStar.description)
+    .setAuthor(message.author.name, message.author.displayAvatarURL)
+    .setTimestamp()
+    .setFooter(`⭐ ${parseInt(star[1])+1} | ${message.id}`)
+    .setImage(image);
+  const starMsg = await message.guild.channels.find('name', starboard).fetchMessage(stars.id);
+  await starMsg.edit({ embed });
+}
+```
+
+Now, if you were to just use the code above, your starboard would only function if there was already a message in the starboard channel. Let's take care of that.
+
+Here we add an if statement that mimics the previous block, but this time manually setting the color of the embed, and also manually setting the amount of stars the embed will have.
+
+```js
+if (!stars) {
+  if (!message.guild.channels.exists('name', starboard)) throw `It appears that you do not have a \`${starboard}\` channel.`;
```

Comment:
I only just noticed, you throw messages, do you account for this in any of the other code? because if I remember correctly this would just log to console instead of responding to the user.